src/index.ts

```diff
@@ -22,16 +31,15 @@ export const create = <T>() => {
       };
     }, [setState]);
 
-    useEffect(() => {
-      sharedState = state;
+    const setStateWrapper: SetState<T> = useCallback(newState => {
       setters.forEach(set => {
-        if (set !== setState) {
-          set(sharedState);
-        }
+        set(prevState => {
+          return isFunction(newState) ? newState(prevState) : newState;
```

Comment:
I would move this out from the component's setState, as your source of truth is `sharedState`, you already have it and you should only need to call `newState` once.

```ts
sharedState = isFunction(newState) ? newState(sharedState) : newState;
setters.forEach(set => set(sharedState));
```

Reply:
actually i updated it now. it longer sets `sharedState`. 
`sharedState` is only used for initializing the state. On subsequent updates, `prevState` is the source of the truth.

I think the implementation is simpler this way